src/ops.rs

```diff
@@ -201,3 +201,48 @@ impl_op_unary! { impl Neg, neg for  Matrix<T, M, N> }
 impl_op_unary! { impl Neg, neg for &Matrix<T, M, N> }
 impl_op_unary! { impl Not, not for  Matrix<T, M, N> }
 impl_op_unary! { impl Not, not for &Matrix<T, M, N> }
+
+////////////////////////////////////////////////////////////////////////////////
+// Matrix * Matrix
+////////////////////////////////////////////////////////////////////////////////
+
+impl<T, const N: usize, const M: usize> Matrix<T, N, M> {
+    pub(crate) fn matmul_into<const P: usize>(
```

Comment:
Also happy for this to be a naked function just in this module. I would prefer naming it something like `matrix_mul`